sweep/sweeper.go

```diff
@@ -1464,6 +1483,57 @@ func (s *UtxoSweeper) handleNewInput(input *sweepInputMessage) {
 	pi.ntfnRegCancel = cancel
 }
 
+// attachAvailableRBFInfo queries the mempool to see whether the given input
+// has already been spent. If so, it will query the sweeper store to fetch the
+// fee info of the spending transction, hence preparing for possible RBF.
+func (s *UtxoSweeper) attachAvailableRBFInfo(pi *pendingInput) *pendingInput {
+	// Check if we can find the spending tx of this input in mempool.
+	tx, spent := s.mempoolLookup(*pi.OutPoint())
+
+	// Exit early if it's not found.
+	//
+	// NOTE: this is not accurate for neutrino backend.
+	if !spent {
+		return pi
+	}
+
+	// Otherwise the input is already spent in the mempool, update its
+	// state to StatePublished.
+	pi.state = StatePublished
+
+	// We also need to update the RBF info for this input. If the sweeping
+	// transaction is broadcast by us, we can find the fee info in the
+	// sweeper store.
+	txid := tx.TxHash()
+	tr, err := s.cfg.Store.GetTx(txid)
+
+	// If the tx is not found in the store, it means it's not broadcast by
+	// us, hence we can't find the fee info. This is fine as, later on when
+	// this tx is confirmed, we will remove the input from our
+	// pendingInputs.
+	if errors.Is(err, ErrTxNotFound) {
+		log.Warnf("Spending tx %v not found in sweeper store", txid)
+		return pi
```

Comment:
Q: I wonder if in this case we should mark the input as invalid because otherwise we are going to fight with a potential party driving the price of a potential RBF up ?

Reply:
If the tx is not found in our store, we won't track it, hence no RBF.

Reply:
Right we mark it as published I think we are good then.